src/components/translator/Translator.tsx

```diff
@@ -272,6 +277,11 @@ const Translator = ({ mode: initialMode }: { mode?: Mode }): React.ReactElement
 
   const onTranslate = React.useCallback(() => window.dispatchEvent(new Event(TranslateEvent)), []);
 
+  const swapLangText = () => {
+    setSrcText(tgtText);
+    setTgtText(srcText);
```

Comment:
Let's encapsulate the language switching here as well.

Reply:
Moving the target text to the source makes sense. But, making the source text the target text doesn't make sense since Apertium's translators are not necessarily bijective. That is, just because "foo" in X translates into "bar" in Y doesn't mean "bar" in Y will translate into "foo" in X.

Reply:
Thanks for pointing that out. Yeah, it makes no sense to make the source text the target text